download_and_create_reference_datasets/v02/hail_scripts/write_combined_reference_data_ht.py

```diff
@@ -294,25 +341,42 @@ def join_hts(datasets, coverage_datasets=[], reference_genome='37'):
                                          datasets=hl.dict(included_dataset),
                                          version=VERSION)
     joined_ht.describe()
+    return joined_ht
 
-    output_path = os.path.join(OUTPUT_TEMPLATE.format(genome_version=reference_genome, version=VERSION))
-    print('Writing to %s' % output_path)
 
-    joined_ht.write(os.path.join(output_path))
+def run(args):
+    # hl._set_flags(no_whole_stage_codegen='1') # hail 0.2.78 hits an error on the join, this flag gets around it
 
+    # If there are out-of-memory error, set the environment variable with the following command
+    # $ export PYSPARK_SUBMIT_ARGS="--driver-memory 4G pyspark-shell"
+    # "4G" in the environment variable can be bigger if your computer has a larger memory.
+    hl.init(default_reference='GRCh38', min_block_size=128, master='local[32]')
 
-def run(args):
-    hl._set_flags(no_whole_stage_codegen='1') # hail 0.2.78 hits an error on the join, this flag gets around it 
-    join_hts(['cadd', '1kg', 'mpc', 'eigen', 'dbnsfp', 'topmed', 'primate_ai', 'splice_ai', 'exac',
+    if args.mitochondrial:
+        logger.info('Joining mitochondrail reference datasets.')
+        joined_ht = join_hts(['gnomad_mito', 'mitomap', 'mitimpact', 'hmtvar', 'helix_mito', 'clinvar_mito', 'dbnsfp_mito'],
+                 reference_genome='38')
+        output_path = 'gs://seqr-reference-data/GRCh38/mitochondrial/all_mito_reference_data/combined_reference_data_chrM.ht'
+        logger.info(f'Writing to {output_path}')
+        joined_ht.write(os.path.join(output_path), overwrite=args.force_write)
+        return
+
+    joined_ht = join_hts(['cadd', '1kg', 'mpc', 'eigen', 'dbnsfp', 'topmed', 'primate_ai', 'splice_ai', 'exac',
               'gnomad_genomes', 'gnomad_exomes', 'geno2mp'],
              ['gnomad_genome_coverage', 'gnomad_exome_coverage'],
              args.build,)
 
+    output_path = os.path.join(OUTPUT_TEMPLATE.format(genome_version=args.build, version=VERSION))
+    logger.info(f'Writing to {output_path}')
+    joined_ht.write(os.path.join(output_path))
+
 
 if __name__ == "__main__":
 
     parser = argparse.ArgumentParser()
     parser.add_argument('-b', '--build', help='Reference build, 37 or 38', choices=["37", "38"], required=True)
+    parser.add_argument('-m', '--mitochondrial', help='Create combined mitochondrial reference dataset hail table', action='store_true')
```

Comment:
as I have repeatedly stated, we need a separate script for writing mito data that imports the functions it needs from here. adding extra entries to the config is fine, but thats the only change to make to this file

Reply:
I see what you mean. I can change the `write_combined_mito_reference_data_hts` to call the joined_ht() from `write_combined_reference_data`. I still need to change the `joined_ht()` function a little bit to write to a different output file.